tests/syscall/mount/mount-immutables.bats

```diff
@@ -0,0 +1,596 @@
+  [[ $immutable_ro_mounts == $immutable_ro_mounts_after ]]
+
+  docker_stop ${syscont}
+}
+
+# Testcase #3.
+#
+# Ensure that a read-write immutable mount *can* be remounted as read-only inside
+# the container, and then back to read-only.
+@test "immutable rw mount can be remounted ro" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  local immutable_rw_mounts=$(list_container_rw_mounts ${syscont} "0" "/")
+  run empty_list ${immutable_rw_mounts}
+  [ "$status" -ne 0 ]
+
+  for m in $immutable_rw_mounts; do
+
+    # Remounting /proc or /dev as read-only will prevent docker execs into the
+    # container; skip these.
+    if [[ ${m} =~ "/proc" ]] || [[ ${m} =~ "/proc/*" ]] ||
+       [[ ${m} =~ "/dev" ]] || [[ ${m} =~ "/dev/*" ]]; then
+      continue
+    fi
+
+    printf "\ntesting ro remount of immutable rw mount ${m}\n"
+
+    docker exec ${syscont} sh -c "mount -o remount,bind,ro ${m}"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "mount -o remount,bind,rw ${m}"
+    [ "$status" -eq 0 ]
+  done
+
+  docker_stop ${syscont}
+}
+
+# Testcase #4.
+#
+# Ensure that a read-only immutable mount *can* be remounted as read-only inside
+# the container.
+@test "immutable ro mount can be remounted ro" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  local immutable_ro_mounts=$(list_container_ro_mounts ${syscont} "0" "/")
+  run empty_list ${immutable_ro_mounts}
+  [ "$status" -ne 0 ]
+
+  for m in $immutable_ro_mounts; do
+    printf "\ntesting ro remount of immutable ro mount ${m}\n"
+
+    docker exec ${syscont} sh -c "mount -o remount,bind,ro ${m}"
+    [ "$status" -eq 0 ]
+  done
+
+  local immutable_ro_mounts_after=$(list_container_ro_mounts ${syscont} "0" "/")
+  [[ $immutable_ro_mounts == $immutable_ro_mounts_after ]]
+
+  docker_stop ${syscont}
+}
+
+# Testcase #5.
+#
+# Ensure that a read-write immutable mount *can* be remounted as read-write or
+# read-only inside the container.
+@test "immutable rw mount can be remounted rw" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  local immutable_rw_mounts=$(list_container_rw_mounts ${syscont} "0" "/")
+  run empty_list ${immutable_rw_mounts}
+  [ "$status" -ne 0 ]
+
+  for m in $immutable_rw_mounts; do
+    printf "\ntesting rw remount of immutable rw mount ${m}\n"
+
+    docker exec ${syscont} sh -c "mount -o remount,bind,rw ${m}"
+    [ "$status" -eq 0 ]
+  done
+
+  local immutable_rw_mounts_after=$(list_container_rw_mounts ${syscont} "0" "/")
+  [[ $immutable_rw_mounts == $immutable_rw_mounts_after ]]
+
+  docker_stop ${syscont}
+}
+
+# Testcase #6.
+#
+# Ensure that a read-only immutable mount can't be bind-mounted
+# to a new mountpoint then re-mounted read-write
```

Comment:
Fix comment: 

" Ensure that a read-only immutable mount  can be bind-mounted to a new mountpoint, but not re-mounted read-write at the new mountpoint."